code/modules/projectiles/projectile.dm

```diff
@@ -106,6 +106,7 @@ var/list/impact_master = list()
 	var/rotate = 1 //whether the projectile is rotated based on angle or not
 	var/superspeed = 0 //When set to 1, the projectile will travel at twice the normal speed
 	var/super_speed = 0 //This exists just for proper functionality
+	var/travel_range = 0	//if set, the projectile will be deleted when its distance from the firing location exceeds this
```

Comment:
Does kill_count get used anywhere? If not, could you remove it please since you're basically replacing it.

If yes, you can just leave it or whatever.